ui/components/app/collectible-details/collectible-details.js

```diff
@@ -270,31 +276,49 @@ export default function CollectibleDetails({ collectible }) {
             >
               {t('contractAddress')}
             </Typography>
-            <Typography
-              color={COLORS.PRIMARY1}
-              variant={TYPOGRAPHY.H6}
-              overflowWrap={OVERFLOW_WRAP.BREAK_WORD}
-              boxProps={{
-                margin: 0,
-                marginBottom: 4,
-              }}
-              className="collectible-details__contract-link"
+            <Box
+              display={DISPLAY.FLEX}
+              flexDirection={FLEX_DIRECTION.ROW}
+              className="collectible-details__contract-wrapper"
             >
-              <a
-                target="_blank"
-                rel="noopener noreferrer"
-                href={getTokenTrackerLink(
-                  address,
-                  currentNetwork,
-                  null,
-                  null,
-                  rpcPrefs,
-                )}
-                title={address}
+              <Typography
+                color={COLORS.PRIMARY1}
+                variant={TYPOGRAPHY.H6}
+                overflowWrap={OVERFLOW_WRAP.BREAK_WORD}
+                boxProps={{
+                  margin: 0,
+                  marginBottom: 4,
+                }}
+                className="collectible-details__contract-link"
               >
-                {inPopUp ? shortenAddress(address) : address}
-              </a>
-            </Typography>
+                <a
+                  target="_blank"
+                  rel="noopener noreferrer"
+                  href={getTokenTrackerLink(
+                    address,
+                    currentNetwork,
+                    null,
+                    null,
+                    rpcPrefs,
+                  )}
+                  title={address}
+                >
+                  {inPopUp ? shortenAddress(address) : address}
+                </a>
+              </Typography>
+              <div
+                className="collectible-details__contract-copy-button"
+                onClick={() => {
+                  handleCopy(address);
+                }}
+              >
```

Comment:
Above should be button. It can be something simple as [this](https://github.com/MetaMask/metamask-extension/blob/develop/ui/components/app/edit-gas-fee-button/index.scss#L6).

It will be nicer to have this kind of buttons in ui-kit cc @georgewrmarshall

Reply:
Will create a ticket(#13404) for an `IconButton` component. Please let me know of you run into any other components that you find yourself creating frequently!